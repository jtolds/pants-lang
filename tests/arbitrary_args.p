func = {|x, *(y)|
  print x "*"
  y @each {|x| print x}
}

func 1 2 3 4

print.

func = {|*(left); *(right)|

  print "left:"
  left @each {|x| print x}

  print "right:"
  right @each {|x| print x}

  print.
}

func.
1 2 func. 3
"hey" func.
"hey" "there" @func "dude" "this" "is" "cool"
func "whoa" "nelly" 1 2 3

func = {|x, y, *(left);|
  print "x:" x
  print "y:" y
  left @each {|z| print z}
}

1 2 3 4 @func
