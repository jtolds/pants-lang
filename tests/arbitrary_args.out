1 *
2
3
4

left:
right:

left:
1
2
right:
3

left:
hey
right:

left:
hey
there
right:
dude
this
is
cool

left:
right:
whoa
nelly
1
2
3

x: 1
y: 2
3
4

return code: 0
